default.nix

```
@@ -157,21 +157,15 @@ let
         '';
       };
 
-    # Tests the tool on the pinned Nixpkgs tree, this is a good sanity check
-    nixpkgsCheck =
-      pkgs.runCommand "test-nixpkgs-check-by-name"
-        {
-          nativeBuildInputs = [
-            packages.build
-            pkgs.nix
-          ];
-          nixpkgsPath = nixpkgs;
-        }
-        ''
-          ${initNix}
-          nixpkgs-check-by-name --base "$nixpkgsPath" "$nixpkgsPath"
-          touch $out
-        '';
+    # Tests the tool on the pinned Nixpkgs tree with stable Nix. This is a good sanity check.
+    # There's a `passthru` attribute that allows testing this on every other Nix version in nixpkgs.
+    nixpkgsCheck = pkgs.callPackage ./test-nixpkgs-check-by-name.nix {
+      inherit initNix nixpkgs;
+      nixpkgs-check-by-name = packages.build;
+    };
+
+    # Tests the tool on the pinned Nixpkgs tree with the most recent Nix. This is a sanity check.
+    nixpkgsCheckWithLatestNix = packages.nixpkgsCheck.passthru.latest;
```

```suggestion
    nixpkgsCheckWithLatestNix = packages.nixpkgsCheck.nixVersions.latest;
```

`passthru`'s get "passed through" as a derivation attribute, so there's no need to refer to it via `passthru`

---

The use of the inner `nixVersions` attrset resolved the reason I was using `passthru` directly here.

---

(not sure what you mean here or why it's resolved)